src/RemoteRequest/CurlRemoteGetRequest.php

```diff
@@ -97,13 +98,17 @@ public function __construct($sslVerify = true, $timeout = self::DEFAULT_TIMEOUT,
     public function get($url, $headers = [])
     {
         $retriesLeft = $this->retries;
+
+        if (! is_string($url) || empty($url)) {
+            throw FailedToGetFromRemoteUrl::withException($url, FailedToParseUrl::forUrl($url));
+        }
```

Comment:
This is good. 

Do you think it would be overkill to also check the `$url` with `parse_url()`? I suppose so since cURL itself would fail and return a better error code in that case?

Reply:
We can add `parse_url()` if we want to avoid a curl request with the wrong URL and throw the error beforehand. 

> I suppose so since cURL itself would fail and return a better error code in that case?

cURL fails but the exception looks like this. I think we can add it with a more clear message like `Passed string is not a valid URL`.

![image](https://user-images.githubusercontent.com/54371619/196851409-3cecc0e9-7307-4a07-8dd8-7e8365342fd9.png)

Reply:
I think this is good. The error coming from cURL will be more descriptive.